### DISCUSSION TITLE
Core-command calls does not publish data on messagebus

### DISCUSSION THREAD

**Discussion by teenccu** [edgexfoundry/community#186] [Q&A] [answered]
Hello,

If I am not mistaken in the configuration Edgex 3.0, I noticed that only auto-events on the device publishes data on the messagebus however any REST call to core-command to read a resource (https://host:8443/core-command/api/v3/device/name/devicename/deviceresource) does not publish an event on the messagebus.

If that is the case , then the question is there a reason for this design?
And is there is a way to change the behavior ?

Thanks for your valuable response.

**Comment by cloudxxx8** [comment, marked as answer]
@teenccu In fact, core command can trigger the event publish via the query parameter.
https://docs.edgexfoundry.org/3.0/api/core/Ch-APICoreCommand/
You can set `ds-pushevent=true`, for example `https://host:8443/core-command/api/v3/device/name/devicename/deviceresource?ds-pushevent=true`
![image](https://github.com/edgexfoundry/community/assets/11695863/6092fbf7-0a25-41b0-bdc7-a49a5de3de2b)
